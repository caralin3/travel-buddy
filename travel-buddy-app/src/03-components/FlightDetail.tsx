import moment from 'moment';
import React from 'react';
import { Link } from 'react-router-dom';
import { Col, Row } from 'reactstrap';
import { Flight } from '../api';
import { TRIPS_ROUTE } from '../router';

export interface FlightDetailProps {
  flight: Flight;
  showTrip?: boolean;
}

export const FlightDetail: React.FC<FlightDetailProps> = ({
  flight: {
    airline,
    arrivalAirport,
    arrivalCity,
    arrivalDate,
    confirmationCode,
    departureAirport,
    departureCity,
    departureDate,
    flightNumber,
    seats,
    terminal,
    trip,
  },
  showTrip = false,
}) => {
  return (
    <div className="flight-detail">
      {showTrip && (
        <Row>
          <Col>
            <Link className="text-dark" to={`${TRIPS_ROUTE}/${trip.id}`}>
              {trip.title}
            </Link>
          </Col>
        </Row>
      )}
      <Row>
        <Col className="flight-detail__name text-primary">
          {airline} #{flightNumber}
        </Col>
        <Col className="flight-detail__cities">
          {departureCity} to {arrivalCity}
        </Col>
      </Row>
      <Row className="my-2">
        <Col xs={6} md={3}>
          <p className="flight-detail__label">Depart</p>
          {moment(departureDate).format('h:mm A')}
        </Col>
        <Col xs={6} md={3}>
          <p className="flight-detail__label">Airport</p>
          {departureAirport}
        </Col>
        <Col xs={6} md={3}>
          <p className="flight-detail__label">Confirmation</p>
          {confirmationCode}
        </Col>
        <Col xs={6} md={3}>
          <p className="flight-detail__label">Notes</p>
          Terminal {terminal}
          {!!seats ? `; Seats ${seats}` : ''}
        </Col>
      </Row>
      <Row>
        <Col xs={6} md={3}>
          <p className="flight-detail__label">Arrive</p>
          {moment(arrivalDate).format('h:mm A')}
        </Col>
        <Col xs={6} md={3}>
          <p className="flight-detail__label">Airport</p>
          {arrivalAirport}
        </Col>
      </Row>
    </div>
  );
};
