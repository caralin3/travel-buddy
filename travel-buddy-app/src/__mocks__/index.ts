export * from './activities';
export * from './cruises';
export * from './flights';
export * from './hotels';
export * from './trips';
export * from './user';
