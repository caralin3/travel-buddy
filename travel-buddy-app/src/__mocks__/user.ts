import { User } from '../api';

export const user: User = {
  id: 0,
  firstName: 'Jane',
  lastName: 'Doe',
  email: 'janeD@email.com',
  roles: [
    {
      id: 0,
      name: 'ROLE_USER',
    },
  ],
};
