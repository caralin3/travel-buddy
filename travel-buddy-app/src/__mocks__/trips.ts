import { Trip } from '../api';
import { user } from './user';

export const trips: Trip[] = [
  {
    user,
    id: 0,
    title: 'Texas Girls Trip 2022',
    description: 'Travelers: Cara, Elizabeth, Sarah<br/>Destination: Dallas, Texas',
    startDate: '2022-06-03',
    endDate: '2022-06-06',
    uniqueLink: '',
  },
  {
    user,
    id: 1,
    title: 'Trip 2',
    startDate: '2022-08-14',
    endDate: '2022-08-31',
  },
];
