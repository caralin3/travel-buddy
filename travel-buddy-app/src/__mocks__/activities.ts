import { Activity } from '../api';
import { trips } from './trips';

export const activities: Activity[] = [
  {
    id: 0,
    name: 'Horseback Riding',
    description: 'Horseback riding at Stockyard stables.',
    startDate: '2022-06-04 11:00:00',
    endDate: '2022-06-04 12:00:00',
    cost: 75,
    currency: 'USD',
    addressLine1: '157 E Exchange Ave',
    city: 'Fort Worth',
    state: 'Texas',
    country: 'USA',
    postalCode: '76164',
    trip: trips[0],
  },
  {
    id: 1,
    name: 'Billy Bobs Texas',
    description: 'Honky Tonk near Stockyard stables.',
    startDate: '2022-06-04 16:00:00',
    endDate: '2022-06-04 18:00:00',
    cost: 20,
    currency: 'USD',
    addressLine1: '2520 Rodeo Plaza',
    city: 'Fort Worth',
    state: 'Texas',
    country: 'USA',
    postalCode: '76164',
    trip: trips[0],
  },
  {
    id: 2,
    name: 'The Sixth Floor Museum at Dealey Plaza',
    description: 'JFK Museum',
    startDate: '2022-06-05 13:30:00',
    endDate: '2022-06-05 15:00:00',
    cost: 1,
    currency: 'USD',
    addressLine1: '411 Elm Street',
    city: 'Dallas',
    state: 'Texas',
    country: 'USA',
    postalCode: '75202',
    trip: trips[0],
  },
  {
    id: 3,
    name: "Adair's Saloon",
    description:
      'Same area=Dallas World Aquarium, 6th floor museum and dealy plaza, and Museum of Nature and Science SE of hotel ',
    startDate: '2022-06-03 16:00:00',
    endDate: '2022-06-03 18:00:00',
    cost: 20,
    currency: 'USD',
    addressLine1: '2624 Commerce Street',
    city: 'Dallas',
    state: 'Texas',
    country: 'USA',
    postalCode: '75226',
    trip: trips[0],
  },
];
