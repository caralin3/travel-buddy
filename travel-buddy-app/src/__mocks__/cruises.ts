import { Cruise, Port } from '../api';
import { trips } from './trips';

export const cruises: Cruise[] = [
  {
    id: 0,
    cabinType: 'SUITE',
    cruiseLine: 'Princess Cruises',
    shipName: 'Discovery Princess',
    cost: 3000,
    currency: 'USD',
    cabinNumber: '11234',
    startDate: '2022-05-08 16:00:00',
    endDate: '2022-05-15 07:00:00',
    description: 'Alaskan Cruise',
    departureCity: 'Seattle',
    departureState: 'Washington',
    departureCountry: 'US',
    roundTrip: true,
    trip: trips[0],
  },
  {
    id: 1,
    cabinType: 'DOUBLE',
    cruiseLine: 'Royal Caribbean',
    shipName: 'Discovery Princess',
    cost: 5402.97,
    currency: 'USD',
    cabinNumber: '9112',
    startDate: '2025-04-12',
    endDate: '2025-04-19',
    departureCity: 'Sicily',
    departureCountry: 'Italy',
    destinationCity: 'Athens',
    destinationCountry: 'Greece',
    roundTrip: false,
    trip: trips[0],
  },
];

export const cruise1Ports: Port[] = [
  {
    id: 0,
    day: 2,
    arrival: '2022-05-10 06:30:00',
    departure: '2022-05-10 15:00:00',
    city: 'Ketchikan',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 1,
    day: 3,
    arrival: '2022-05-11 12:30:00',
    departure: '2022-05-11 21:00:00',
    city: 'Juneau',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 2,
    day: 4,
    arrival: '2022-05-12 06:00:00',
    departure: '2022-05-12 17:00:00',
    city: 'Skagway',
    state: 'Alaska',
    country: 'US',
    cruise: cruises[0],
  },
  {
    id: 3,
    day: 6,
    arrival: '2022-05-14 19:00:00',
    departure: '2022-05-14 23:59:00',
    city: 'Victoria',
    country: 'Canada',
    cruise: cruises[0],
  },
];

export const cruise2Ports: Port[] = [
  {
    id: 0,
    day: 2,
    arrival: '2025-04-14 06:30:00',
    departure: '2025-04-14 15:00:00',
    city: 'Florence',
    country: 'Italy',
    cruise: cruises[1],
  },
  {
    id: 1,
    day: 5,
    arrival: '2025-04-17 12:30:00',
    departure: '2025-04-17 21:00:00',
    city: 'Rome',
    country: 'Italy',
    cruise: cruises[1],
  },
  {
    id: 2,
    day: 6,
    arrival: '2025-04-18 06:00:00',
    departure: '2025-04-18 17:00:00',
    city: 'Santorini',
    country: 'Greece',
    cruise: cruises[1],
  },
];
