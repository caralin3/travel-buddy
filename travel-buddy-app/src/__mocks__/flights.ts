import { Flight } from '../api';
import { trips } from './trips';

export const flights: Flight[] = [
  {
    id: 1,
    airline: 'JetBlue',
    arrivalAirport: 'JFK',
    arrivalCity: 'NYC',
    arrivalDate: '2022-06-03 15:30:00',
    confirmationCode: 'FWD5DBJ',
    departureAirport: 'DFW',
    departureCity: 'Dallas',
    departureDate: '2022-06-06 13:05:00',
    flightClass: 'ECONOMY',
    flightNumber: '124',
    seats: '12 DEF',
    terminal: 'E',
    trip: trips[0],
  },
  {
    id: 0,
    airline: 'JetBlue',
    arrivalAirport: 'DFW',
    arrivalCity: 'Dallas',
    arrivalDate: '2022-06-03 13:39:00',
    confirmationCode: 'FWD5DBJ',
    cost: 300.03,
    currency: 'USD',
    departureAirport: 'JFK',
    departureCity: 'NYC',
    departureDate: '2022-06-03 10:05:00',
    flightClass: 'ECONOMY',
    flightNumber: '123',
    terminal: '5',
    seats: '7 ABC',
    trip: trips[0],
  },
];
