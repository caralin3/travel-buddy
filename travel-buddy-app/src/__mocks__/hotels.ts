import { Hotel } from '../api';
import { trips } from './trips';

export const hotels: Hotel[] = [
  {
    id: 0,
    name: 'Courtyard By Marriot Denton',
    roomType: 'DOUBLE',
    roomCount: 1,
    checkInDate: '2022-06-03 15:00:00',
    checkOutDate: '2022-06-06 11:00:00',
    cost: 0,
    currency: 'USD',
    addressLine1: '2800 Colorado Blvd',
    city: 'Denton',
    state: 'TX',
    trip: trips[0],
  },
];
