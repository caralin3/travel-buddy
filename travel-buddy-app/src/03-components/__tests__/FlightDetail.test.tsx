import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { flights } from '../../__mocks__';
import { FlightDetail } from '../FlightDetail';

describe('FlightDetail', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<FlightDetail flight={flights[0]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render with trip correctly', () => {
    const wrapper = shallow(<FlightDetail showTrip flight={flights[1]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
