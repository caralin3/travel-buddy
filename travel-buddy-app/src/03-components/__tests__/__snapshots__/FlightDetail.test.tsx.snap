// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightDetail should render correctly 1`] = `
<div
  className="flight-detail"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      className="flight-detail__name text-primary"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      JetBlue
       #
      124
    </Col>
    <Col
      className="flight-detail__cities"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      Dallas
       to 
      NYC
    </Col>
  </Row>
  <Row
    className="my-2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Depart
      </p>
      1:05 PM
    </Col>
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Airport
      </p>
      DFW
    </Col>
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Confirmation
      </p>
      FWD5DBJ
    </Col>
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Notes
      </p>
      Terminal 
      E
      ; Seats 12 DEF
    </Col>
  </Row>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Arrive
      </p>
      3:30 PM
    </Col>
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Airport
      </p>
      JFK
    </Col>
  </Row>
</div>
`;

exports[`FlightDetail should render with trip correctly 1`] = `
<div
  className="flight-detail"
>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <Link
        className="text-dark"
        to="/trips/0"
      >
        Texas Girls Trip 2022
      </Link>
    </Col>
  </Row>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      className="flight-detail__name text-primary"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      JetBlue
       #
      123
    </Col>
    <Col
      className="flight-detail__cities"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      NYC
       to 
      Dallas
    </Col>
  </Row>
  <Row
    className="my-2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Depart
      </p>
      10:05 AM
    </Col>
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Airport
      </p>
      JFK
    </Col>
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Confirmation
      </p>
      FWD5DBJ
    </Col>
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Notes
      </p>
      Terminal 
      5
    </Col>
  </Row>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Arrive
      </p>
      1:39 PM
    </Col>
    <Col
      md={3}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
      xs={6}
    >
      <p
        className="flight-detail__label"
      >
        Airport
      </p>
      DFW
    </Col>
  </Row>
</div>
`;
